modules/extensions-support/src/main/java/org/mule/module/extension/internal/introspection/describer/MuleExtensionAnnotationParser.java

```diff
@@ -268,4 +272,22 @@ private static boolean isDisplayAnnotationPresent(AnnotatedElement annotatedElem
         return displayAnnotations.stream().anyMatch(annotation -> annotatedElement.getAnnotation(annotation) != null);
     }
 
+    /**
+     * Enriches the {@link ParameterDescriptor} with a {@link MetadataModelProperty} if the parsedParameter is
+     * annotated either as {@link Content} or {@link MetadataKeyParam}
+     *
+     * @param parsedParameter the method annotated parameter parsed
+     * @param parameter       the {@link ParameterDescriptor} associated to the parsed parameter
+     */
+    public static void parseMetadataAnnotations(AnnotatedElement parsedParameter, ParameterDescriptor parameter)
+    {
+        if (parsedParameter.getAnnotation(Content.class) != null)
```

Comment:
add a unit test which asserts that this model property is correctly set for each of these cases